### PR TITLE
sinkv2(ticdc): add more sinkv2 metrics

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: ref #5928 

### What is changed and how it works?

Improve sinkv2 metrics code and add panels.

#### Tests 

 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sinkv2/metrics/metrics.go [hunk 1]
```diff
@@ -14,71 +14,70 @@
...
-	TxnBatchSize = prometheus.NewHistogram(
+	TxnWorkerFlushDuration = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
 			Namespace: "ticdc",
-			Subsystem: "sink",
-			Name:      "txn_batch_size",
-			Help:      "Size of the DML transaction batch.",
-		})
+			Subsystem: "sinkv2",
+			Name:      "txn_worker_flush_duration",
```
Comment (hi-rustin) [inline cdc/sinkv2/metrics/metrics.go:40]:
```suggestion
			Name:      "txn_worker_flush_duration.",
```

File: cdc/sinkv2/metrics/metrics.go [hunk 2]
```diff
@@ -14,71 +14,70 @@
...
+			Help:      "Flush duration (s) for txn worker",
+			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20), // 1ms~1000s
+		}, []string{"namespace", "changefeed"})
+
+	TxnWorkerBusyRatio = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "ticdc",
+			Subsystem: "sinkv2",
+			Name:      "txn_worker_busy_ratio",
+			Help:      "Busy ratio (X ms in 1s) for all workers",
```
Comment (hi-rustin) [inline cdc/sinkv2/metrics/metrics.go:50]:
```suggestion
			Help:      "Busy ratio (X ms in 1s) for all workers.",
```

File: cdc/sinkv2/eventsink/txn/worker.go [hunk 3]
```diff
@@ -104,7 +117,11 @@ func (w *worker) runBackgroundLoop() {
 			zap.Int("workerID", w.ID))
 
 		w.timer = time.NewTimer(w.flushInterval)
-		wantMoreCallbacks := make([]func(), 0, 1024)
+		w.wantMoreCallbacks = make([]func(), 0, 1024)
+
+		var flushTimeSlice, totalTimeSlice time.Duration
+		overseerTimer := time.NewTicker(2 * time.Second)
```
Comment (hi-rustin) [inline cdc/sinkv2/eventsink/txn/worker.go:123]: When to close it?

File: cdc/sinkv2/eventsink/txn/worker.go [hunk 4]
```diff
@@ -161,9 +189,16 @@ func (w *worker) doFlush(needStopTimer bool) bool {
...
 	}
-	if needStopTimer && !w.timer.Stop() {
-		<-w.timer.C
+
+	// Flush successfully, call callbacks to notify conflict detector.
+	for _, wantMore := range w.wantMoreCallbacks {
+		wantMore()
+	}
+	w.wantMoreCallbacks = w.wantMoreCallbacks[:0]
+	if cap(w.wantMoreCallbacks) > 1024 {
```
Comment (hi-rustin) [inline cdc/sinkv2/eventsink/txn/worker.go:198]: Does it help GC? Better to add some comments.

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

**Review by hi-rustin — APPROVED** [review]
Approved in advance because it was all nit issues.

**Comment by hi-rustin** [comment]
@hicqu Before you submit the PR please follow [this doc](https://github.com/pingcap/tiflow/blob/master/CONTRIBUTING.md#format-of-the-commit-message) to organize your PR title. Thanks!

File: cdc/sinkv2/eventsink/txn/worker.go [hunk 5]
```diff
@@ -148,8 +158,27 @@ func (w *worker) runBackgroundLoop() {
 	}()
 }
 
+func (w *worker) onEvent(txn txnWithNotifier) bool {
+	w.metricConflictDetectDuration.Observe(time.Since(txn.start).Seconds())
+	w.wantMoreCallbacks = append(w.wantMoreCallbacks, txn.wantMore)
+	if w.backend.OnTxnEvent(txn.txnEvent.TxnCallbackableEvent) {
+		if !w.timer.Stop() {
```
Comment (nongfushanquan) [inline cdc/sinkv2/eventsink/txn/worker.go:165]: what's  the purpose of this timer

File: cdc/sinkv2/eventsink/txn/worker.go [hunk 6]
```diff
@@ -119,27 +137,19 @@ func (w *worker) runBackgroundLoop() {
...
-			for _, wantMore := range wantMoreCallbacks {
-				wantMore()
-			}
-			wantMoreCallbacks = wantMoreCallbacks[:0]
-			if cap(wantMoreCallbacks) > 1024 {
-				wantMoreCallbacks = make([]func(), 0, 1024)
+			case now := <-overseerTimer.C:
+				totalTimeSlice = now.Sub(startToWork)
+				busyRatio := int(flushTimeSlice.Seconds() / totalTimeSlice.Seconds() * 1000)
+				w.metricTxnWorkerBusyRatio.Add(float64(busyRatio) / float64(w.workerCount))
```
Comment (nongfushanquan) [inline cdc/sinkv2/eventsink/txn/worker.go:150]: why use the workercount as a denominator here

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**